milestone3/frontend/src/components/RecipePage.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ChangeEvent, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom'
+import recipes from '../recipeData';
+import './RecipePage.css'
+
+interface RecipePageProps {
+    external: boolean;
+}
+
+export default function RecipePage(props: RecipePageProps) {
+    const { name, id } = useParams();
+    const defaultRecipe = recipes[0];
+    const [recipe, setRecipe] = useState(defaultRecipe)
+    const [newIngredient, setNewIngredient] = useState('');
+    const [allIngredients, setAllIngredients] = useState(defaultRecipe.ingredients)
+    const [newInstruction, setNewInstruction] = useState('');
+    const [allInstructions, setAllInstructions] = useState(defaultRecipe.instructions)
+
+    useEffect(() => {
+        if (props.external) {
+            fetch(`https://bootcamp-milestone-4.onrender.com/recipe/${name}`)
+                .then(res => res.json())
+                .then(data => {setRecipe(data[0])})
```

Comment:
Try to add a .catch() to the end of your .then() chains so you can catch and print any errors!